app/src/features/sessionBook/screens/DraftSessionScreen/components/SessionTrimmer/SessionTrimmer.tsx

```diff
@@ -0,0 +1,167 @@
+import React, { useCallback, useRef, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { getSessionRecordingEvents } from "store/features/session-recording/selectors";
+import { RQButton } from "lib/design-system/components";
+import "./sessionTrimmer.scss";
+import { EventType, eventWithTime } from "rrweb";
+import { fullSnapshotEvent, IncrementalSource } from "@rrweb/types";
+import { NetworkEventData, RQSessionEventType } from "@requestly/web-sdk";
+import { sessionRecordingActions } from "store/features/session-recording/slice";
+import { partition } from "lodash";
+import { Divider } from "antd";
+import { TrimHandle } from "./components/TrimHandle/TrimHandle";
+import { msToMinutesAndSeconds } from "utils/DateTimeUtils";
+
+interface SessionTrimmerProps {
+  duration: number;
+  sessionStartTime: number;
+}
+
+export const SessionTrimmer: React.FC<SessionTrimmerProps> = ({ duration, sessionStartTime }) => {
+  const dispatch = useDispatch();
+  const [startTime, setStartTime] = useState(0);
+  const [endTime, setEndTime] = useState(duration);
+  const containerRef = useRef<HTMLDivElement>(null);
+  const [isSessionTrimmed, setIsSessionTrimmed] = useState(false);
+  const events = useSelector(getSessionRecordingEvents);
+
+  const handleDrag = useCallback(
+    (event: MouseEvent, isStart: boolean) => {
+      if (!containerRef.current) return;
+      const rect = containerRef.current.getBoundingClientRect();
+      const x = event.clientX - rect.left;
+      const newTime = Math.round((x / rect.width) * duration);
+      if (!isSessionTrimmed) {
+        setIsSessionTrimmed(true);
+      }
+      if (isStart) {
+        setStartTime(Math.max(0, Math.min(newTime, endTime - 1)));
+      } else {
+        setEndTime(Math.min(duration, Math.max(newTime, startTime + 1)));
+      }
+    },
+    [duration, startTime, endTime]
+  );
+
+  const startDrag = useCallback(
+    (isStart: boolean) => {
+      const handleMouseMove = (e: MouseEvent) => handleDrag(e, isStart);
+      const handleMouseUp = () => {
+        document.removeEventListener("mousemove", handleMouseMove);
+        document.removeEventListener("mouseup", handleMouseUp);
+      };
+
+      document.addEventListener("mousemove", handleMouseMove);
+      document.addEventListener("mouseup", handleMouseUp);
+    },
+    [handleDrag]
+  );
+
+  const handleDiscardTrim = () => {
+    setIsSessionTrimmed(false);
+    setStartTime(0);
+    setEndTime(duration);
+    dispatch(sessionRecordingActions.setTrimmedSessiondata(null));
+  };
+
+  const handleTrim = useCallback(() => {
```

Comment:
Can we move the trimming logic into a separate function so that logic is not interfered with unintentionally?

```js
handleTrim(){
// trimSessionEvents()
//state/component specific actions
}
```